### ISSUE TITLE
az should prevent user from generating sas with a friendly message

### ISSUE THREAD

**Issue by lzomlot** [Azure/azure-cli#9078]
The issue has been copied to your clipboard. Paste it here!
To print out the issue body locally, run `a

## Describe the bug

**Command Name**
`az storage blob generate-sas`

**Errors:**

:open_mouth:

```The command failed with an unexpected error. Here is the traceback:

self.account_key should not be None.Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\knack\knack\cli.py", line 206, in invoke
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 326, in execute
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 384, in _run_jobs_serially
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 375, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-storage\azure\cli\command_modules\storage\__init__.py", line 240, in new_handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-storage\azure\cli\command_modules\storage\__init__.py", line 183, in handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 354, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 145, in __call__
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\__init__.py", line 451, in default_command_handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-storage\azure\cli\command_modules\storage\operations\blob.py", line 423, in generate_sas_blob_uri
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-multiapi-storage\azure\multiapi\storage\v2018_03_28\blob\baseblobservice.py", line 494, in generate_blob_shared_access_signature
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-multiapi-storage\azure\multiapi\storage\v2018_03_28\common\_error.py", line 130, in _validate_not_none
ValueError: self.account_key should not be None.
```

## To Reproduce:
Steps to reproduce the behavior. Note: Command arguments have been redacted.

- `Fill in additional info here`
- `Run: az storage blob generate-sas -c {} -n {} --auth-mode {} --account-name {}`

## Environment Summary
```
Windows-10-10.0.17763-SP0
Python 3.6.6
Shell: powershell.exe

azure-cli                         2.0.62

```
## Additional Context

z feedback --verbose`

**Comment by yugangw-msft** [comment]
@limingu, can you please fix this by sprint 64 with a nice error?

**Comment by limingu** [comment]
#9546

**Comment by limingu** [comment]
This issue has been fixed in version 2.0.67.
